### PR TITLE
redcoding asset handler, holder, tracker, memory asset

### PR DESCRIPTION
**Pull request by carlitosan**
Signed-off-by: carlitosan <82187351+carlitosan@users.noreply.github.com>

### CODE REVIEW

File: Gems/ScriptCanvas/Code/Editor/View/Widgets/StatisticsDialog/NodeUsageTreeItem.h [hunk 1]
```diff
@@ -94,15 +94,16 @@ namespace ScriptCanvasEditor
...
         void OnAssetReady(AZ::Data::Asset<AZ::Data::AssetData> asset) override;
         void OnAssetSaved(AZ::Data::Asset<AZ::Data::AssetData> asset, bool isSuccessful) override;
         void OnAssetReloaded(AZ::Data::Asset<AZ::Data::AssetData> asset) override;
+        */
         ////
-
-    private:
-
-        void ProcessAsset(const AZ::Data::Asset<ScriptCanvas::ScriptCanvasAssetBase>& scriptCanvasAsset);
+        // void ProcessAsset(const AZ::Data::Asset<LegacyScriptCanvasBaseAsset>& scriptCanvasAsset);
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/Editor/View/Widgets/StatisticsDialog/NodeUsageTreeItem.h:106]: You can delete all this now :)

File: Gems/ScriptCanvas/Code/Editor/View/Widgets/StatisticsDialog/ScriptCanvasStatisticsDialog.cpp [hunk 2]
```diff
@@ -176,29 +176,32 @@ namespace ScriptCanvasEditor
...
-        {
-            m_scriptCanvasAssetTreeRoot->RegisterAsset(assetId, assetInfo.m_assetType);
-        }
+        // #sc_editor_asset_redux cut or update
+//         AZ::Data::AssetInfo assetInfo;
+//         AZ::Data::AssetCatalogRequestBus::BroadcastResult(assetInfo, &AZ::Data::AssetCatalogRequestBus::Events::GetAssetInfoById, assetId);
+//         if (assetInfo.m_assetType == azrtti_typeid<OldAssetType())
+//         {
+//             m_scriptCanvasAssetTreeRoot->RegisterAsset(assetId, assetInfo.m_assetType);
+//         }
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/Editor/View/Widgets/StatisticsDialog/ScriptCanvasStatisticsDialog.cpp:187]: delete

File: Gems/ScriptCanvas/Code/Editor/View/Widgets/StatisticsDialog/ScriptCanvasStatisticsDialog.cpp [hunk 3]
```diff
@@ -176,29 +176,32 @@ namespace ScriptCanvasEditor
...
-            m_scriptCanvasAssetTreeRoot->RemoveAsset(assetId);
-        }
+        // #sc_editor_asset_redux cut or update
+
+//         AZ::Data::AssetInfo assetInfo;
+//         AZ::Data::AssetCatalogRequestBus::BroadcastResult(assetInfo, &AZ::Data::AssetCatalogRequestBus::Events::GetAssetInfoById, assetId);
+//         if (assetInfo.m_assetType == azrtti_typeid<OldAssetType>())
+//         {
+//             m_scriptCanvasAssetTreeRoot->RemoveAsset(assetId);
+//         }
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/Editor/View/Widgets/StatisticsDialog/ScriptCanvasStatisticsDialog.cpp:204]: delete

File: Gems/ScriptCanvas/Code/Editor/View/Widgets/StatisticsDialog/ScriptCanvasStatisticsDialog.cpp [hunk 4]
```diff
@@ -432,22 +435,23 @@ namespace ScriptCanvasEditor
...
+//                 const AzToolsFramework::AssetBrowser::ProductAssetBrowserEntry* productEntry = static_cast<const AzToolsFramework::AssetBrowser::ProductAssetBrowserEntry*>(entry);
+// 
+//                 if (productEntry->GetAssetType() == azrtti_typeid<OldAssetType>())
+//                 {
+//                     const AZ::Data::AssetId& assetId = productEntry->GetAssetId();
+// 
+//                     m_scriptCanvasAssetTreeRoot->RegisterAsset(assetId, productEntry->GetAssetType());
+//                 }
+//             }
+//         }
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/Editor/View/Widgets/StatisticsDialog/ScriptCanvasStatisticsDialog.cpp:454]: delete

File: Gems/ScriptCanvas/Code/Editor/View/Widgets/UnitTestPanel/UnitTestDockWidget.cpp [hunk 5]
```diff
@@ -26,7 +26,7 @@
 #include <AzCore/Serialization/EditContext.h>
 #include <AzCore/IO/FileIO.h>
 #include <AzCore/Asset/AssetManager.h>
-
+#include <Editor/Assets/ScriptCanvasAssetHelpers.h>
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/Editor/View/Widgets/UnitTestPanel/UnitTestDockWidget.cpp:29]: organize along with the other Editor/ includes

File: Gems/ScriptCanvas/Code/Editor/View/Widgets/UnitTestPanel/UnitTestDockWidget.cpp [hunk 6]
```diff
@@ -47,13 +46,12 @@
...
 
-#include <ScriptCanvas/Assets/ScriptCanvasAsset.h>
-#include <ScriptCanvas/Assets/ScriptCanvasAssetHandler.h>
+
 #include <ScriptCanvas/Bus/ScriptCanvasExecutionBus.h>
 #include <ScriptCanvas/Bus/UnitTestVerificationBus.h>
 #include <ScriptCanvas/Data/DataRegistry.h>
 #include <ScriptCanvas/GraphCanvas/NodeDescriptorBus.h>
-
+#include <ScriptCanvas/Components/EditorUtils.h>
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/Editor/View/Widgets/UnitTestPanel/UnitTestDockWidget.cpp:54]: alphabetize the order

File: Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp [hunk 7]
```diff
@@ -3513,19 +3493,23 @@ namespace ScriptCanvasEditor
...
+    AZ::EntityId MainWindow::FindEditorNodeIdByAssetNodeId([[maybe_unused]] const ScriptCanvasEditor::SourceHandle& assetId
+        , [[maybe_unused]] AZ::EntityId assetNodeId) const
     {
-        AZ::EntityId editorEntityId;
-        AssetTrackerRequestBus::BroadcastResult
-            ( editorEntityId, &AssetTrackerRequests::GetEditorEntityIdFromSceneEntityId, assetId.Id(), assetNodeId);
+        AZ::EntityId editorEntityId{};
+//         AssetTrackerRequestBus::BroadcastResult
+//             ( editorEntityId, &AssetTrackerRequests::GetEditorEntityIdFromSceneEntityId, assetId.Id(), assetNodeId);
+        // #sc_editor_asset_redux fix logger
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp:3502]: delete

File: Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp [hunk 8]
```diff
@@ -3513,19 +3493,23 @@ namespace ScriptCanvasEditor
...
+    AZ::EntityId MainWindow::FindAssetNodeIdByEditorNodeId([[maybe_unused]] const ScriptCanvasEditor::SourceHandle& assetId
+        , [[maybe_unused]] AZ::EntityId editorNodeId) const
     {
-        AZ::EntityId sceneEntityId;
-        AssetTrackerRequestBus::BroadcastResult
-            ( sceneEntityId, &AssetTrackerRequests::GetSceneEntityIdFromEditorEntityId, assetId.Id(), editorNodeId);
+        AZ::EntityId sceneEntityId{};
+        // AssetTrackerRequestBus::BroadcastResult
+        // ( sceneEntityId, &AssetTrackerRequests::GetSceneEntityIdFromEditorEntityId, assetId.Id(), editorNodeId);
+        // #sc_editor_asset_redux fix logger
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp:3512]: delete

File: Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Core.h [hunk 9]
```diff
@@ -317,6 +317,24 @@ namespace ScriptCanvasEditor
...
+        inline static constexpr const char* GetDescription() { return "Script Canvas Graph File"; }
+        inline static constexpr const char* GetSuggestedSavePath() { return "@projectroot@/scriptcanvas"; }
+        inline static constexpr const char* GetFileExtension() { return ".scriptcanvas"; }
+        inline static constexpr const char* GetGroup() { return "Script Canvas"; }
+        inline static constexpr const char* GetAssetNamePattern() { return "Untitled-%i"; }
+        inline static constexpr const char* GetFileFilter() { return "Script Canvas Files (*.scriptcanvas)"; }
+        inline static constexpr const char* GetAssetTypeDisplayName() { return "Script Canvas"; }
+        inline static constexpr const char* GetEntityName() { return "Script Canvas"; }
+        inline static constexpr const char* GetIconPath() { return "Icons/ScriptCanvas/Viewport/ScriptCanvas.png"; }
+        inline static AZ::Color GetDisplayColor() { return AZ::Color(0.5f, 0.5f, 0.5f, 0.5f); };
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Core.h:335]: this should be coming from the style sheet, we'll have to address it in a future change

**Review by lsemp3d — APPROVED** [review]
Some minor cleanup is needed, but otherwise looks great!

**Review by cgalvan — APPROVED** [review]

**Pull request merged**